tasks/insights-client.yml

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: MIT
+---
+- name: Ensure required packages are installed
+  package:
+    name: insights-client
+    state: present
+  when: rhc_insights.state | d("present") == "present"
```

Comment:
I think that, since all the tasks here need to be run when `rhc_insights.state=present`, maybe we can apply the condition to the `include_tasks: insights-client.yml` in `main.yml`; since this helper file is used to register to insights, we can assume this in all the tasks, and simplify things a bit